android/modules/ui/src/java/ti/modules/titanium/ui/widget/tableview/TiTableView.java

```diff
@@ -68,6 +72,45 @@ public TiTableView(TableViewProxy proxy)
 		this.recyclerView.setBackgroundColor(Color.TRANSPARENT);
 		this.recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
 
+		// Add listener to fire scroll events.
+		this.recyclerView.addOnScrollListener(new RecyclerView.OnScrollListener()
+		{
+			@Override
+			public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newState)
+			{
+				super.onScrollStateChanged(recyclerView, newState);
+
+				if (newState == RecyclerView.SCROLL_STATE_IDLE) {
+					final KrollDict payload = generateScrollPayload();
+					final TiNestedRecyclerView nestedRecyclerView = getRecyclerView();
+
+					// Obtain last touch position for `scrollend` event.
+					final TiDimension xDimension =
+						new TiDimension(nestedRecyclerView.getLastTouchX(), TiDimension.TYPE_WIDTH);
+					final TiDimension yDimension =
+						new TiDimension(nestedRecyclerView.getLastTouchY(), TiDimension.TYPE_HEIGHT);
+					payload.put(TiC.EVENT_PROPERTY_X, xDimension.getAsDefault(nestedRecyclerView));
+					payload.put(TiC.EVENT_PROPERTY_Y, yDimension.getAsDefault(nestedRecyclerView));
+
+					proxy.fireSyncEvent(TiC.EVENT_SCROLLEND, payload);
+				}
+			}
+
+			@Override
+			public void onScrolled(@NonNull RecyclerView recyclerView, int dx, int dy)
+			{
+				super.onScrolled(recyclerView, dx, dy);
+
+				// Update scroll offsets.
+				scrollOffsetX += dx;
+				scrollOffsetY += dy;
+
+				final KrollDict payload = generateScrollPayload();
+
+				proxy.fireSyncEvent(TiC.EVENT_SCROLL, payload);
+			}
+		});
+
```

Comment:
We document that `TableView`'s "scroll" event provides properties `firstVisibleItem`, `totalItemCount`, and `visibleItemCount`. All 3 of these event properties return values in Titanium 9.2.2, but with this PR only `firstVisibleItem` is supported. The `totalItemCount` and `visibleItemCount` event properties are undefined.
https://docs.appcelerator.com/platform/latest/#!/api/Titanium.UI.TableView-event-scroll

Side Note:
iOS does not support these 3 event properties. They're undefined on iOS.